lesson06/task3/server/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+var name string
+var address string
+
+func handler(w http.ResponseWriter, r *http.Request) {
+
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
+		return
+	}
+
+	switch r.Method {
+	case "GET":
+
+		// GET return HTML page
+		http.ServeFile(w, r, "form.html")
+
+	case "POST":
+
+		if err := r.ParseForm(); err != nil {
+			fmt.Fprintf(w, "ParseForm() err: %v", err)
+			return
+		}
+
+		name = r.FormValue("name")
+
+		address = r.FormValue("address")
+
+		fmt.Fprintf(w, "%s from %s\n", name, address)
+		fmt.Printf("%s from %s\n", name, address)
+
+	default:
+		fmt.Fprintf(w, "Sorry, only GET and POST methods are supported.")
+	}
+}
+
+func docHandler(w http.ResponseWriter, r *http.Request) {
+
+	expire := time.Now().Add(200 * time.Minute)
+	cookie := &http.Cookie{
+		Name:    "token#1",
+		Value:   fmt.Sprintf("%s:%s", name, address),
+		MaxAge:  300,
+		Expires: expire,
+	}
+
+	http.SetCookie(w, cookie)
+
+	w.WriteHeader(200)
+	w.Write([]byte(fmt.Sprintf("%s Get Successful\n", cookie.Name)))
+
+	//Print Cookie
+
+	fmt.Println(cookie)
+
+	return
+}
+
+func main() {
+	var port int = 8080
+	fmt.Printf("Server started at port %d\n", port)
+	http.HandleFunc("/", handler)
+	docHandler := http.HandlerFunc(docHandler)
+	http.Handle("/doc", docHandler)
```

Comment:
Homework description doesn't say anything about serving anything for `/doc` or anything specific for the appropriate handler. The cookie should be used on `/` path to pass the form values

Reply:
For me "reload page" means to do another GET request. In that case I need to do another handler for GET req. To solve that, i made req. as well as hadler for another path.  What is wrong with that? Please give me right direction!

Reply:
I hope Cookie is fixed.

Reply:
Maybe the wording was not really clear, and the word "reload" threw you off. The requirement was to make a handler that serves a page. The html was actually given and you can use the one provided in the description. This page is accessed on `/` with GET and POST requests. You have a form on the page and obviously when you try to load the page in your browser it is being rendered by a GET handler. The page has a form with two inputs for name and address, the submit button will trigger a POST request, which should serve the same page, but in this case the values of the name and address should be saved into a cookie and when the page renders again on this same POST request (this is what was implied by "reload", which, I agree, is not really a correct term) these values for name and address are pulled from this cookie and are displayed on this page along the same form with name and address fields. Essentially both GET and POST requests need to render the same page with the difference, that POST also has to do some work with passing values through the cookie.
You have another handler that serves a page on `/doc`, which was not in the homework description, this is what I indicated in the comment above.
I hope this helps.